action.yml

```
@@ -0,0 +1,55 @@
+name: 'Build Plugin'
+description: 'Build GOG Galaxy integration on current OS'
+
+inputs:
+  src:
+    description: 'Location of directory containing manifest.json and essential plugin files'
+    required: true
+    default: 'src'
+  requirements:
+    description: 'Location of Python requirements file'
+    required: true
+    default: 'requirements/app.txt'
+  output-zip-name:
+    description: 'Name of created zip that contains build plugin'
+    required: true
+    default: 'build.zip'
+  third-party-target:
+    description: 'Target directory where pip packages are dumped; relative to resulted build'
+    required: false
+    default: '.'
+  python-arch:
+    description: 'Python binary to be used: x64 for Mac or x86 for Windows'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.7.5'  # Version 3.7.3 used by GOG not avaialable https://github.com/actions/setup-python/issues/474
```

```suggestion
        python-version: '3.7.5'  # Version 3.7.3 used by GOG not avaialable https://github.com/actions/setup-python/issues/474
        cache: 'pip' # caching pip dependencies
```

Should we cache pip dependencies?

---

Does it cache from build to build? I'm not sure if it will change anything as fog.buildtools is using `piptools` under the hood to generate temporary requirements file which THEN is used by pip to install stuff. I woudn't optimize prematuraly if it works ok

---

It caches the pip packages but I think it uses the requirements.txt so probably best to just leave it our after we’ve done some more in-depth research.